id.go

```diff
@@ -97,14 +100,14 @@ func init() {
 		dec[encoding[i]] = byte(i)
 	}
 
-	// If PID is 1 and /proc/1/cpuset exists and is not /, we can assume that we
+	// If /proc/1/cpuset exists and is not /, we can assume that we
 	// are in a form of container and use the content of /proc/1/cpuset instead
 	// of the PID.
-	if pid == 1 {
-		b, err := ioutil.ReadFile("/proc/1/cpuset")
-		if err == nil && len(b) > 1 {
-			pid = int(crc32.ChecksumIEEE(b))
-		}
+	// See http://man7.org/linux/man-pages/man7/cpuset.7.html.
+	b, err := ioutil.ReadFile("/proc/1/cpuset")
+	if err == nil && len(b) > 1 {
```

Comment:
So basically, every process within the same cpuset will have the same pid. I'm not sure it is the right solution.

Also, you should use `/proc/self/cpuset` now.

Reply:
We could xor the crc32 with the pid.

Reply:
Nice review.